src/parser.js

```diff
@@ -1001,6 +1001,40 @@ function insertSuperCalls(ast) {
   return ast;
 }
 
+function isArrayAssignmentWithThisReference(path) {
+  const assignedPath = path.value && path.value.left;
+  const isArrayAssignment = assignedPath && assignedPath.type === 'ArrayExpression';
+  const hasMemberExpressions = isArrayAssignment && jsc(assignedPath).find('MemberExpression').size() > 0;
+
+  return isArrayAssignment && hasMemberExpressions;
+}
+
+function insertArrayAssignmentVarDeclarations(assignmentPath) {
+  const identifiers = jsc(assignmentPath.value.left)
+    .find(jsc.Identifier)
+    .filter(path => path.parentPath.name === 'elements')
+    .paths()
+    .map(path => path.value.name)
+    .map(identifierName => jsc.variableDeclarator(jsc.identifier(identifierName), null));
```

Comment:
It's possible, for sure.  For me, having them separate creates a nice functional flow of transformations.  Unpacking three operations in one line is a fair amount of cognitive load that I'd like to avoid.

Compromise on this?
```
.map(path => jsc.identifier(path.value.name)
.map(identifier => jsc.variableDeclarator(identifier, null));
```